File: SDKs/monarchinitiative.org/1.0.1/python/src/sdk/models/operations/get_relation_usage_resource.py
import dataclasses
from typing import Optional
from ..shared import associationresults as shared_associationresults


@dataclasses.dataclass
class GetRelationUsageResourceQueryParams:
    evidence: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'evidence', 'style': 'form', 'explode': True }})
    subject_taxon: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'subject_taxon', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class GetRelationUsageResourceRequest:
    query_params: GetRelationUsageResourceQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class GetRelationUsageResourceResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    association_results: Optional[list[shared_associationresults.AssociationResults]] = dataclasses.field(default=None)
    

File: SDKs/monarchinitiative.org/1.0.1/typescript/src/sdk/models/operations/getrelationusageresource.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRelationUsageResourceQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=evidence" })
  evidence?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=subject_taxon" })
  subjectTaxon?: string;
}


export class GetRelationUsageResourceRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetRelationUsageResourceQueryParams;
}


export class GetRelationUsageResourceResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ elemType: shared.AssociationResults })
  associationResults?: shared.AssociationResults[];

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
